### PR TITLE
Add support to custom weapon item name and add weapon tint functionality

### PR DESCRIPTION
**Pull request by pstomi**
1. Added support to able to use custom weapon item name like instead of WEAPON_SMG we can use item name smg. Many servers use this way and the current hardcoded WEAPON_ solution prevent them to use ox_inventory, because it would be really painful for them to migrate their existing database with custom weapon names to use new ox_inventory. 

2. Added functionality for the weapon tints, so servers wont need separate script for that. 

3. Bugfix for item effects

### CODE REVIEW

File: web/src/components/inventory/InventoryContext.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ const InventoryContext: React.FC<{
           <Item onClick={handleClick} data={{ action: 'drop' }}>
             {Locale.ui_drop}
           </Item>
-          {props.item.name.startsWith('WEAPON_') && props.item.metadata?.serial && (
+          {props.item.metadata?.serial && (
```
Comment (LukeWasTakenn) [inline web/src/components/inventory/InventoryContext.tsx:68]: How'd this work? Not all weapons have a serial 🤔

**Comment by thelindat** [comment]
> Many servers use this way and the current hardcoded WEAPON_ solution prevent them to use ox_inventory, because it would be really painful for them to migrate their existing database

I don't really care.

File: web/src/components/inventory/InventoryContext.tsx (hunk 1 above)
Comment (pstomi) [reply web/src/components/inventory/InventoryContext.tsx]: Fixed, I added new change where it checks with hashname. @LukeWasTakenn

**Comment by pstomi** [comment]
> > Many servers use this way and the current hardcoded WEAPON_ solution prevent them to use ox_inventory, because it would be really painful for them to migrate their existing database
> 
> I don't really care.

@thelindat   I know it is not your problem, but here is the solution what would help many people. Also in general it is never a good thing to use hardcoded stuffs in an actual code, it is not a good coding practice, it can lead to many problem and a code is harder to maintain.

**Comment by thelindat** [comment]
> I know it is not your problem

Literally everything people try to push is my problem.

**Comment by pstomi** [comment]
> > I know it is not your problem
> 
> Literally everything people try to push is my problem.

@thelindat  so no chance that this will be approved and merged?

**Comment by thelindat** [comment]
I need to fix various issues I have with this or at least review piece by piece for you to fix.

**Comment by thelindat** [comment]
I've implemented the functionality you wanted, so you should just need to PR your tints.

**Comment by pstomi** [comment]
@thelindat   Thank you so much. I will open separate PR about the Tints.

**Pull request closed without merging**